src/runner.c

```diff
@@ -47,15 +49,26 @@ void handleEvents(
       instance->tail = event;
     }
   }
-  uv_async_send(instance->async);
   uv_mutex_unlock(&instance->mutex);
+  uv_async_send(&instance->async);
```

Comment:
Is this necessary? In the async+data structure scenario, it's considered good form to notify with the lock held, for the same reason you call pthread_cond_signal() with the mutex held.